supervisor/addons/addon.py

```diff
@@ -1530,7 +1530,7 @@ async def _restart_after_problem(self, state: ContainerState):
                 except AddonsError as err:
                     attempts = attempts + 1
                     _LOGGER.error("Watchdog restart of addon %s failed!", self.name)
-                    capture_exception(err)
+                    await self.sys_run_in_executor(capture_exception, err)
```

Comment:
I was actually wondering that, is `capture_exception` really blocking I/O? I guess it is, but it nowhere really says, from what I can tell? :thinking: 

In any case, I am a bit worried that we don't get some very fatal issues anymore with this e.g. what if the error leads to a segfault right afterwards? In that case the Sentry wouldn't make it out.

Now I am not sure, maybe that is a bit scripted, and given that people who opt-in to sentry might actually block unnecessarily in their production system, we should probably really do this in an executor.

But I'd like to have this centralized. Can you create a `async_capture_exception` which calls the executor? This would allow us to "opt-out", e.g. for a dev instance or something.

Reply:
> I was actually wondering that, is capture_exception really blocking I/O? I guess it is, but it nowhere really says, from what I can tell? 🤔

It is. Though not necessarily because of Sentry. I also am not sure if the library i blocking I/O or not. But it doesn't seem to support async and makes HTTP calls so I think it is?

But the reason I know its I/O is because of us. We add read the remaining free space from the disk every time and attach that info as metadata in the call here:
https://github.com/home-assistant/supervisor/blob/696dcf6149840f1217ab4c04406280d42219eee9/supervisor/misc/filter.py#L83

That was how it ended up in part of this PR. I found the `read_text` call in the `free_space` property and backtracked to where it was used.